### PR TITLE
True OpenAI drop-in replacement by InferenceClient

### PR DESCRIPTION
**Pull request by Wauplin**
Closes https://github.com/huggingface/huggingface_hub/issues/2369 @lappemic 

Goal is to be able to use `InferenceClient` exactly the same way as `OpenAI` client. To do so we need to:
- [x] rename `model` to `base_url` => added an alias for it
- [x] rename `model_id` to `model`
- [x] rename `token` to `api_key` => added an alias for it
- [x] add alias for `client.chat.completions.create`

@philschmid could you have a look at it and confirm it meets your expectations? See tests for real example ([here](https://github.com/huggingface/huggingface_hub/pull/2384/files#diff-53ed5db45d648482e7373b2291d45216a2dae782cd864772f8de717793c9cbfa) and [here](https://github.com/huggingface/huggingface_hub/pull/2384/files#diff-76a73351bb753f100ea324d6594ac33bacb13b1068a88c70a15daf4314470cc5))

### Sync + `stream=False`
```py
client = InferenceClient(
    base_url="https://api-inference.huggingface.co/models/meta-llama/Meta-Llama-3-8B-Instruct",
    api_key="my-api-key",
)
output = client.chat.completions.create(
    model="meta-llama/Meta-Llama-3-8B-Instruct",
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Count to 10"},
    ],
    stream=False,
    max_tokens=1024,
)
assert output.choices[0].message.content == "1, 2, 3, 4, 5, 6, 7, 8, 9, 10!"
```

### Sync + `stream=True`

```py
client = InferenceClient()
output = client.chat.completions.create(
    model="meta-llama/Meta-Llama-3-8B-Instruct",
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Count to 10"},
    ],
    stream=True,
    max_tokens=1024,
)

for chunk in output:
    print(chunk.choices[0].delta.content)
```

### Async + `stream=False`

```py
    client = AsyncInferenceClient(
        base_url="https://api-inference.huggingface.co/models/meta-llama/Meta-Llama-3-8B-Instruct",
        api_key="my-api-key",
    )
    output = await client.chat.completions.create(
        model="meta-llama/Meta-Llama-3-8B-Instruct",
        messages=[
            {"role": "system", "content": "You are a helpful assistant."},
            {"role": "user", "content": "Count to 10"},
        ],
        stream=False,
        max_tokens=1024,
    )
    assert output.choices[0].message.content == "1, 2, 3, 4, 5, 6, 7, 8, 9, 10!"
```

### Async + `stream=True`

```py
client = AsyncInferenceClient()
output = await client.chat.completions.create(
    model="meta-llama/Meta-Llama-3-8B-Instruct",
    messages=[
        {"role": "system", "content": "You are a helpful assistant."},
        {"role": "user", "content": "Count to 10"},
    ],
    stream=True,
    max_tokens=1024,
)

chunked_text = [chunk.choices[0].delta.content async for chunk in output]
```

### TODO:

- [x] document this properly (see [inference.md](https://github.com/huggingface/huggingface_hub/pull/2384/files#diff-40fee3ecf18715d547ce9d9f1a0e305aba17bf008033a635f9b62ca05cb45a53))

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
Awesome, very clear docs! Thanks @Wauplin

File: docs/source/en/guides/inference.md [hunk 1]
```diff
@@ -147,6 +147,65 @@ endpoints.
...
+    stream = await client.chat.completions.create(
+        model="meta-llama/Meta-Llama-3-8B-Instruct",
+        messages=[{"role": "user", "content": "Say this is a test"}],
+        stream=True,
+    )
+    async for chunk in stream:
+        print(chunk.choices[0].delta.content or "", end="")
+
+asyncio.run(main())
+```
```
Comment (philschmid) [inline docs/source/en/guides/inference.md:201]: do you want to use a `dif`?

Comment (Wauplin) [reply docs/source/en/guides/inference.md]: Good all, updated in https://github.com/huggingface/huggingface_hub/pull/2384/commits/3e30e6f390fbe09812cb5ce43811cbefd50ec38c

**Review by philschmid — APPROVED** [review]

**Comment by julien-c** [comment]
> @philschmid could you have a look at it and confirm it meets your expectations?

from experience, a very tough test to pass usually 🤣

**Comment by Wauplin** [comment]
Thanks for the reviews!

**Pull request merged**